### PR TITLE
Add 5.5.0 to 5.6.0 migration updates related to fix#1616

### PR DESCRIPTION
**Pull request by vihanga-liyanage**
Proposed changes
- DB scripts to change the table name IDN_AUTH_TEMP_SESSION_DATA_STORE to IDN_AUTH_TEMP_SESSION_STORE
- Migration service to update EXPIRY_TIME by adding _session clean up time_ out to TIME_CREATED

Resolves product-is/issues/3277

### CODE REVIEW

File: modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v560/dao/SessionDAO.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    public void updateSessionExpireTime(Long cleanUpTimeOut, Connection connection) throws SQLException {
+
+        try (PreparedStatement preparedStatement = connection.prepareStatement(UPDATE_IDN_AUTH_SESSION_STORE)) {
+            preparedStatement.setLong(1, cleanUpTimeOut);
+            preparedStatement.executeUpdate();
+        }
+    }
+}
```
Comment (senthalan) [inline modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v560/dao/SessionDAO.java]: Please add new line end of the file

File: modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v560/migrator/SessionDataMigrator.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        // Get session clean up time out
+        Long cleanUpTimeOut = IdentityUtil.getCleanUpTimeout();
+
+        try (Connection connection = getDataSource().getConnection()) {
+            SessionDAO.getInstance().updateSessionExpireTime(cleanUpTimeOut, connection);
+        } catch (SQLException e) {
+            throw new MigrationClientException("Error while updating session expiry times. ", e);
+        }
+    }
+}
```
Comment (senthalan) [inline modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v560/migrator/SessionDataMigrator.java]: Please add new line end of the file

File: modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v560/util/SQLConstants.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.is.migration.service.v560.util;
+
+public class SQLConstants {
+
+    public static final String UPDATE_IDN_AUTH_SESSION_STORE = "UPDATE IDN_AUTH_SESSION_STORE SET " +
+            "EXPIRY_TIME = TIME_CREATED + ?";
+}
```
Comment (senthalan) [inline modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v560/util/SQLConstants.java]: Please add new line end of the file

File: modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v560/dao/SessionDAO.java (hunk 1 above)
Comment (vihanga-liyanage) [reply modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v560/dao/SessionDAO.java]: There is a new line at EOF. Git diff just doesn't show it. If it wasn't, it'll show a red arrow. :)

File: modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v560/migrator/SessionDataMigrator.java (hunk 2 above)
Comment (vihanga-liyanage) [reply modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v560/migrator/SessionDataMigrator.java]: There is a new line at EOF.

File: modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v560/util/SQLConstants.java (hunk 3 above)
Comment (vihanga-liyanage) [reply modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v560/util/SQLConstants.java]: There is a new line at EOF.

File: modules/migration/migration-resources/5.6.0/dbscripts/step1/identity/db2.sql [hunk 4]
```diff
@@ -30,3 +31,5 @@ CREATE TRIGGER SP_CLAIM_DIALECT_TRIG NO CASCADE
 /
 ALTER TABLE SP_CLAIM_DIALECT ADD CONSTRAINT DIALECTID_APPID_CONSTRAINT FOREIGN KEY (APP_ID) REFERENCES SP_APP (ID) ON DELETE CASCADE
 /
+ALTER TABLE IDN_AUTH_SESSION_STORE ADD EXPIRY_TIME BIGINT AFTER TIME_CREATED;
```
Comment (malithie) [inline modules/migration/migration-resources/5.6.0/dbscripts/step1/identity/db2.sql]: As the column is added to the last no need to specify to add after 'TIME_CREATED' column right

File: modules/migration/migration-resources/5.6.0/dbscripts/step1/identity/h2.sql [hunk 5]
```diff
@@ -19,3 +20,5 @@ CREATE TABLE IF NOT EXISTS SP_CLAIM_DIALECT (
 );
 
 ALTER TABLE SP_CLAIM_DIALECT ADD CONSTRAINT DIALECTID_APPID_CONSTRAINT FOREIGN KEY (APP_ID) REFERENCES SP_APP (ID) ON DELETE CASCADE;
+
+ALTER TABLE IDN_AUTH_SESSION_STORE ADD EXPIRY_TIME BIGINT AFTER TIME_CREATED;
```
Comment (malithie) [inline modules/migration/migration-resources/5.6.0/dbscripts/step1/identity/h2.sql]: As the column is added to the last no need to specify to add after 'TIME_CREATED' column right

File: modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v560/migrator/SessionDataMigrator.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    @Override
+    public void migrate() throws MigrationClientException {
+
+        // Get session clean up time out
+        Long cleanUpTimeOut = IdentityUtil.getCleanUpTimeout();
+
+        try (Connection connection = getDataSource().getConnection()) {
+            SessionDAO.getInstance().updateSessionExpireTime(cleanUpTimeOut, connection);
+        } catch (SQLException e) {
+            throw new MigrationClientException("Error while updating session expiry times. ", e);
```
Comment (malithie) [inline modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v560/migrator/SessionDataMigrator.java]: We had an option to continue upon an error and later re-run the client again. Please have a check and do the needful

File: modules/migration/migration-resources/5.6.0/dbscripts/step1/identity/db2.sql (hunk 4 above)
Comment (vihanga-liyanage) [reply modules/migration/migration-resources/5.6.0/dbscripts/step1/identity/db2.sql]: Fixed

File: modules/migration/migration-resources/5.6.0/dbscripts/step1/identity/h2.sql (hunk 5 above)
Comment (vihanga-liyanage) [reply modules/migration/migration-resources/5.6.0/dbscripts/step1/identity/h2.sql]: Fixed

File: modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v560/migrator/SessionDataMigrator.java (hunk 6 above)
Comment (vihanga-liyanage) [reply modules/migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v560/migrator/SessionDataMigrator.java]: Fixed.

**Pull request merged**